src/components/jobs/edit/jobEditDialog.ts

```diff
@@ -208,11 +208,49 @@ export class JobEditDialog {
         } else {
             this.formState.schedule = true;
         }
-        if (!this.job.offset && this.job.offset !== 0) {
+        if ((!this.job.offset && this.job.offset !== 0) || this.job.offset < 0) {
             error = true;
             this.formState.offset = false;
         } else {
-            this.formState.offset = true;
+            if (this.job.schedule_id) {
+                let job_schedule = null;
```

Comment:
Can we extract this validation logic into a service or some similar abstraction? This is quite a bit to be repeating verbatim.